src/object.d

```diff
@@ -3287,135 +3302,57 @@ int __cmp(T1, T2)(T1[] s1, T2[] s2)
-
-        static int compare(F f1, F f2)
-        {
-            static if (is(F == cfloat) || is(F == cdouble) || is(F == creal))
-            {
-                // Use rt.cmath2._Ccmp instead ?
-                int result;
-
-                if (f1.re < f2.re)
-                    result = -1;
-                else if (f1.re > f2.re)
-                    result = 1;
-                else if (f1.im < f2.im)
-                    result = -1;
-                else if (f1.im > f2.im)
-                    result = 1;
-                else
-                    result = 0;
-                return result;
-            }
-            else static if (is(F == float) || is(F == double) || is(F == real))
-            {
-                // d1 is NaN 2^0 + d2 is NaN 2^1
-                auto NaNs = (f1 != f1) | ((f2 != f2) << 1);
-
-                return (NaNs == 3) ? 0 :
-                    (NaNs == 2) ? 1 :
-                    (NaNs == 1) ? -1 :
-                    (f1 < f2) ? -1 : (f1 > f2);
-            }
-            else static assert(false, "Internal error");
-        }
 
-        auto len = s1.length;
-        if (s2.length < len)
-            len = s2.length;
-
-        auto fpArray1 = () { return cast(F[])s1; }();
-        auto fpArray2 = () { return cast(F[])s2; }();
-
-        foreach (const u; 0 .. len)
-        {
-            if (int c = compare(fpArray1[u], fpArray2[u]))
-                return c;
-        }
-        return s1.length < s2.length ? -1 : (s1.length > s2.length);
+        auto fpArray1 = () @trusted { return cast(F[])s1; }();
+        auto fpArray2 = () @trusted { return cast(F[])s2; }();
+        return __cmp(fpArray1, fpArray2);
     }
-    // char types = > dstrcmp
-    else static if ((is(U1 == ubyte) && is(U2 == ubyte))
-        || (is(U1 == void) && is(U2 == void))
-        || (is(U1 == bool) && is(U2 == bool))
-        || (is(U1 == char) && is(U2 == char)))
+    // All unsigned byte-wide types = > dstrcmp
+    else static if (is(U1 == ubyte) || is(U1 == void) || is(U1 == bool)
+        || is(U1 == char))
     {
-        if (!__ctfe)
-        {
-            import core.internal.string : dstrcmp;
-            return () @trusted { return dstrcmp(cast(char[])s1, cast(char[])s2); }();
-        }
-        else
-        {
-            // pretty ugly...
-            auto len = s1.length;
-
-            if (s2.length < len)
-                len = s2.length;
-
-            foreach (const u; 0 .. len)
-            {
-                auto e1 = at(s1, u);
-                auto e2 = at(s2, u);
-
-                if (e1 < e2)
-                    return -1;
-
-                if (e1 > e2)
-                    return 1;
-            }
-
-            return s1.length < s2.length ? -1 : (s1.length > s2.length);
-        }
+        import core.internal.string : dstrcmp;
+        return (() @trusted => dstrcmp(cast(char[])s1, cast(char[])s2))();
     }
-    // integral, struct, nested arrays
+    // Everything else
```

Comment:
It had become untrue due to the changes in this diff. There are only a few leading special cases for objects and imaginary types, so... "Everything that is not a class/interface or imaginary type..."